storage/innobase/handler/ha_innodb.cc

```diff
@@ -8861,13 +8861,13 @@ ha_innobase::update_row(
 		const bool vers_ins_row = vers_set_fields
 			&& thd_sql_command(m_user_thd) != SQLCOM_ALTER_TABLE;
 
-		/* This is not a delete */
-		m_prebuilt->upd_node->is_delete =
-			(vers_set_fields && !vers_ins_row) ||
-			(thd_sql_command(m_user_thd) == SQLCOM_DELETE &&
-				table->versioned(VERS_TIMESTAMP))
-			? VERSIONED_DELETE
-			: NO_DELETE;
+		if (vers_set_fields && !vers_ins_row) {
+			m_prebuilt->upd_node->is_delete = TRX_ID_DELETE;
+		} else if (thd_sql_command(m_user_thd) == SQLCOM_DELETE) {
+			m_prebuilt->upd_node->is_delete = TIMESTAMP_DELETE;
```

Comment:
Can we avoid adding calls to `thd_sql_command()`?
Do we really have to call it for every non-versioned `update_row()` operation?